rascal2/ui/presenter.py

```diff
@@ -71,6 +75,75 @@ def edit_controls(self, setting: str, value: Any):
 
     def interrupt_terminal(self):
         """Sends an interrupt signal to the terminal."""
-        # TODO: stub for when issue #9 is resolved
-        # https://github.com/RascalSoftware/RasCAL-2/issues/9
-        pass
+        self.runner.interrupt()
+
+    def run(self):
+        """Run RAT."""
+        rat_inputs = RAT.inputs.make_input(self.model.project, self.model.controls)
+        display_on = self.model.controls.display != RAT.utils.enums.Display.Off
+
+        self.runner = RATRunner(rat_inputs, self.model.controls.procedure, display_on)
+        self.runner.finished.connect(self.handle_results)
+        self.runner.stopped.connect(self.handle_interrupt)
+        self.runner.event_received.connect(self.handle_event)
+        self.runner.start()
+
+    def handle_results(self):
+        """Handle a RAT run being finished."""
+        results = self.runner.results
+        self.view.handle_results(results)
+
+    def handle_interrupt(self):
+        """Handle a RAT run being interrupted."""
+        if self.runner.error is None:
+            self.view.logging.info("RAT run interrupted!")
+        else:
+            self.view.logging.critical(f"RAT run failed with exception:\n{self.runner.error}")
```

Comment:
I think this level should be error not critical. Critical should be errors that will lead to a crash or unrecoverable state